vendor/ocaml-4.12/ocaml_412.ml

```diff
@@ -0,0 +1,15 @@
+module Ast_helper = Ast_helper
+module Ast_mapper = Ast_mapper
+module Asttypes = Asttypes
+module Docstrings = Docstrings
+module Let_binding = Let_binding
+module Lexer = Lexer
+module Location = Location
+module Longident = Longident
+module Parse = Parse
+module Parser = Parser
+module Parsetree = Parsetree
+module Pprintast = Pprintast
+module Printast = Printast
```

Comment:
Do we need these modules ? (same for `Warnings`, `Profile`, `Terminfo`)

Reply:
We can remove `Profile` and `Terminfo` indeed. `Warnings` has to be vendored as `Warnings.Unexpected_docstring` has been added in 4.12.

`Printast` and `Pprintast` have to stay in sync with `Parsetree` since their implementation depend on the AST, so they have to stay vendored for now. But they are solely used for debugging I think (I will check) so we can choose to drop these before we merge this PR

Reply:
I think it's fine to keep `Printast` and `Pprintast` as long as we know why. We can also change the debug output to use only one of the two.

Reply:
done in #1671